marge/job.py

```diff
@@ -302,7 +311,14 @@ def push_rebased_and_rewritten_version(
                 trailer_name='Tested-by',
                 trailer_values=tested_by,
                 branch=source_branch,
-                start_commit=source_branch+'^'
+                start_commit=source_branch + '^'
+            )
+        if part_of is not None:
+            rewritten_sha = repo.tag_with_trailer(
+                trailer_name='Part-of',
+                trailer_values=[part_of],
+                branch=source_branch if tested_by is not None else (source_branch + '^'),
```

Comment:
I don't think I like `Part-of`'s tagging behavior depending on whether you also use `Tested-by`. Always tagging the last commit adds some redundancy but also makes trailer based scripting easier and more robust.